app/Http/Controllers/Select/RoleController.php

```diff
@@ -26,21 +26,38 @@
 namespace App\Http\Controllers\Select;
 
 use Illuminate\Http\Request;
-use Silber\Bouncer\BouncerFacade as Bouncer;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Str;
+use Spatie\Permission\Models\Role;
 
 class RoleController extends SelectController
 {
-    protected ?string $idField = 'name';
-    protected ?string $textField = 'title';
-
-    protected function searchFields(Request $request)
+    protected function searchFields(Request $request): array
     {
         return ['name'];
     }
 
-    protected function baseQuery(Request $request)
+    protected function baseQuery(Request $request): \Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder
+    {
+        $this->authorize('viewAny', Role::class);
+
+        if (Role::exists()) {
+            return Role::query()->select('name');
+        }
+
+        // Create a query builder from a raw SQL that returns default values
+        return DB::table(DB::raw("(SELECT 'admin' as name UNION ALL SELECT 'global-read' UNION ALL SELECT 'user') as roles"));
```

Comment:
This is not very nice :(
when do we need this?

Reply:
Yes it is weird, but baseQuery must return a Builder (Eloquent or Fluent).  This way the filtering and sorting works for the select2 instance.  Previously, I just returned a collection for the default roles and phpstan yelled at me :)

Reply:
I'm not sure if we could actually get away with returning a collection here, but that would take deeper changes.